From 99b350df463e03e7a03f203f16ae47a92e7f7810 Mon Sep 17 00:00:00 2001
From: Chuck Crandall <chuck@webchuckweb.com>
Date: Fri, 24 May 2019 10:04:30 -0600
Subject: [PATCH] Updating SQL queries to use LIMIT rather than OFFSET.

---
 lib/UdoitStats.php | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/lib/UdoitStats.php b/lib/UdoitStats.php
index 107fcfe..503cba7 100644
--- a/lib/UdoitStats.php
+++ b/lib/UdoitStats.php
@@ -125,7 +125,8 @@ class UdoitStats
                 $query .= "ORDER BY $date_order DESC\n";
         }
 
-        $query .= "OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
+        //$query .= "OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
+        $query .= "LIMIT $number_items, $offset\n";
 
         $sth = UdoitDB::prepare($query);
         if (isset($get_data['start_date'])) {
@@ -269,7 +270,8 @@ class UdoitStats
                 ."LEFT JOIN $db_reports_table ON($db_user_table.id = $db_reports_table.user_id)\n"
                 ."GROUP BY $db_user_table.id\n"
                 ."ORDER BY $db_user_table.id\n"
-                ."OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
+                //."OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
+                ."LIMIT $number_items, $offset\n";
 
         $sth = UdoitDB::prepare($query);
 
@@ -369,7 +371,8 @@ class UdoitStats
             $query .= $prepend_word." DATE(date_created) <= :enddate\n";
         }
 
-        $query .= "GROUP BY {$date_format} ORDER BY {$date_format} OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY";
+        //$query .= "GROUP BY {$date_format} ORDER BY {$date_format} OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY";
+        $query .= "GROUP BY {$date_format} ORDER BY {$date_format} LIMIT $number_items, $offset";
 
         $sth = UdoitDB::prepare($query);
         if (isset($startDate)) {
-- 
2.20.1

From 756b9835f24db618942a5e2e1c20df38d8eae3a4 Mon Sep 17 00:00:00 2001
From: Chuck Crandall <chuck@webchuckweb.com>
Date: Fri, 24 May 2019 10:29:07 -0600
Subject: [PATCH] Fixing incorrect LIMIT syntax.

---
 lib/UdoitStats.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/UdoitStats.php b/lib/UdoitStats.php
index 503cba7..491fff5 100644
--- a/lib/UdoitStats.php
+++ b/lib/UdoitStats.php
@@ -126,7 +126,7 @@ class UdoitStats
         }
 
         //$query .= "OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
-        $query .= "LIMIT $number_items, $offset\n";
+        $query .= "LIMIT $offset, $number_items\n";
 
         $sth = UdoitDB::prepare($query);
         if (isset($get_data['start_date'])) {
@@ -271,7 +271,7 @@ class UdoitStats
                 ."GROUP BY $db_user_table.id\n"
                 ."ORDER BY $db_user_table.id\n"
                 //."OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY\n";
-                ."LIMIT $number_items, $offset\n";
+                ."LIMIT $offset, $number_items\n";
 
         $sth = UdoitDB::prepare($query);
 
@@ -372,7 +372,7 @@ class UdoitStats
         }
 
         //$query .= "GROUP BY {$date_format} ORDER BY {$date_format} OFFSET $offset ROWS FETCH NEXT $number_items ROWS ONLY";
-        $query .= "GROUP BY {$date_format} ORDER BY {$date_format} LIMIT $number_items, $offset";
+        $query .= "GROUP BY {$date_format} ORDER BY {$date_format} LIMIT $offset, $number_items";
 
         $sth = UdoitDB::prepare($query);
         if (isset($startDate)) {
-- 
2.20.1

From 9033a8497cee94dc943751568f68d397d8fddfcf Mon Sep 17 00:00:00 2001
From: Chuck Crandall <chuck@webchuckweb.com>
Date: Fri, 24 May 2019 10:56:05 -0600
Subject: [PATCH] Fixing hard coded table name in SQL.

---
 lib/UdoitStats.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/UdoitStats.php b/lib/UdoitStats.php
index 491fff5..c35910a 100644
--- a/lib/UdoitStats.php
+++ b/lib/UdoitStats.php
@@ -263,7 +263,7 @@ class UdoitStats
             $date_format = "TO_CHAR(date_created, 'Month dd, YYYY HH:MI:SS AM TZ') AS \"Date Created\", ";
         }
         $query = "SELECT $db_user_table.id as \"User ID\", "
-                ."COUNT(reports.user_id) AS \"Number of Scans\", "
+                ."COUNT($db_reports_table.user_id) AS \"Number of Scans\", "
                 .$date_format
                 ."canvas_url AS \"Canvas URL\"\n"
                 ."FROM $db_user_table\n"
-- 
2.20.1

