diff --git a/public/assets/js/default.js b/public/assets/js/default.js
index f750f3c..e6f01ba 100755
--- a/public/assets/js/default.js
+++ b/public/assets/js/default.js
@@ -390,7 +390,9 @@ $doc.ready(function() {
 
 	// result panel collapsing
 	$doc.on('click', '.panel-heading .btn-toggle', function () {
-		$(this).children('button span').removeClass('glyphicon-minus').addClass('glyphicon-plus');
+		$('span.glyphicon', this).removeClass('glyphicon-minus').addClass('glyphicon-plus');
+		$('span.sr-only span', this).text('Expand');
+
 		var $errorItem = $(this).parent();
 		if ($errorItem.parent().find('.errorSummary').is(':visible')) {
 			$errorItem.parent().find('.errorSummary').slideUp(function () {
@@ -400,7 +402,8 @@ $doc.ready(function() {
 			});
 		}
 		else {
-			$(this).children('button span').removeClass('glyphicon-plus').addClass('glyphicon-minus');
+			$('span.glyphicon', this).removeClass('glyphicon-plus').addClass('glyphicon-minus');
+			$('span.sr-only span', this).text('Collapse');
 			$errorItem.parent().find('.errorSummary').slideDown(function () {
 				resizeFrame();
 			});
diff --git a/templates/partials/results_scannable.php b/templates/partials/results_scannable.php
index e862561..fb40dd7 100644
--- a/templates/partials/results_scannable.php
+++ b/templates/partials/results_scannable.php
@@ -1,4 +1,5 @@
 <?php
+
 /**
  *	Copyright (C) 2014 University of Central Florida, created by Jacob Bates, Eric Colon, Fenel Joseph, and Emily Sachs.
  *
@@ -31,7 +32,10 @@
 		<?php if ($item->amount > 0) : ?>
 			<div class="errorItem panel panel-default">
 				<div class="panel-heading clearfix">
-					<button class="btn btn-xs btn-default btn-toggle pull-left no-print margin-right-small"><span class="glyphicon glyphicon-plus"></span></button>
+					<button class="btn btn-xs btn-default btn-toggle pull-left no-print margin-right-small">
+						<span class="glyphicon glyphicon-plus"></span>
+						<span class="sr-only"><span>Expand</span> <?= $content_group->title . ': ' . $item->name; ?></span>
+					</button>
 
 					<h3 class="plus pull-left"><a class="report-url" href="<?= $item->url; ?>" target="_blank"><?= $item->name; ?>&nbsp;<small><span class="glyphicon glyphicon-new-window"></span></small></a></h3>
 

